spec.md

```diff
@@ -201,6 +205,10 @@ A useful diagram is provided [here](https://github.com/google/go-containerregist
 A registry MAY reject a manifest of any type uploaded to the manifest endpoint if it references manifests or blobs that do not exist in the registry.
 When a manifest is rejected for this reason, it must result in one or more `MANIFEST_BLOB_UNKNOWN` errors <sup>[code-1](#error-codes)</sup>.
 
+The client SHOULD include a `Docker-Manifest-Tag` header indicating which manifest tag is being pushed.
```

Comment:
Same as above, change Docker to OCI, and use a reference rather than the tag. I'd also consider dropping this since the reference may not be known in a lot of cases (the digest of the manifest requires calculating the digest of all the blobs, which may happen during blob upload rather than before all blobs are uploaded). If we keep it, I'd push to change SHOULD to MAY on blob upload.

Reply:
OK, got it - using OCI in the header name, and using full ref instead of only tag.

I agree about not knowing the digest, because it is calculated after/during blobs upload. But the tag should be known when the image is pushed by tag. And as far as I know, at least with Docker client, push is only supported by tag, not by digest. So I think is it ok to ask the client to send `OCI-Reference` with image tag when an image is pushed. 

Personally I prefer asking with "_client SHOULD include_", because in my opinion it increases the possibility for it to be implemented, but "_MAY_" is fine as well.